File: app/src/main/java/com/pedrosoares/cotacaoapp/data/remote/service/Requester.java
package com.pedrosoares.cotacaoapp.data.remote.service;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class Requester {

    private final Retrofit retrofit;

    public Requester() {

        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

        OkHttpClient.Builder client = new OkHttpClient.Builder();
        client.addInterceptor(interceptor);
        String URL = "https://economia.awesomeapi.com.br/";

        retrofit = new Retrofit.Builder()
                .baseUrl(URL)
                .addConverterFactory(GsonConverterFactory.create())
                //.addConverterFactory(GsonConverterFactory.create(buildGson()))
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .client(client.build())
                .build();
    }

    public Api getCoins() {
        return retrofit.create(Api.class);
    }
}

File: app/src/main/java/com/pedrosoares/cotacaoapp/data/remote/service/Requester.kt
package com.pedrosoares.cotacaoapp.data.remote.service

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

object Requester {
    private const val URL = "https://economia.awesomeapi.com.br/"
    private var retrofit: Retrofit? = null
    @JvmStatic
    val service: Retrofit
        get() {
            val interceptor = HttpLoggingInterceptor()
            interceptor.level = HttpLoggingInterceptor.Level.BODY
            val client = OkHttpClient.Builder()
            client.addInterceptor(interceptor)
            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                        .baseUrl(URL)
                        .addConverterFactory(GsonConverterFactory.create())
                        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                        .client(client.build())
                        .build()
            }
            return retrofit!!
        }
}


